src/main/java/io/github/rabobank/shadow_tool/ShadowFlow.java

```diff
@@ -107,9 +107,10 @@ public T compare(final Supplier<T> currentFlow, final Supplier<T> newFlow) {
      * @param newFlow     A supplier that return the result of the new service call
      *                    that you want to start using.
      * @param clazz       The model that the current and new flow should be mapped to for comparison.
+     * @param <C>         The type of collection to compare, for example a List
      * @return This will always return the value of currentFlow supplier.
      */
-    public Collection<T> compareCollections(final Supplier<Collection<T>> currentFlow, final Supplier<Collection<T>> newFlow, final Class<T> clazz) {
+    public <C extends Collection<T>> C compareCollections(final Supplier<C> currentFlow, final Supplier<C> newFlow, final Class<T> clazz) {
```

Comment:
Should we create some tests for this? Currently we only check if some logs are produced, nothing about the types

Reply:
I added some, please have another look

Reply:
See my new comment